lib/statement_formatter.rb

```diff
@@ -0,0 +1,37 @@
+class StatementFormatter
+
+  def create_statement(transactions)
```

Comment:
Really nice and easy to understand! 

It feels like it has two responsibilities mixed togther: calculating the balance and generating the line strings.  Could they be moved to separate methods?